### PR TITLE
[#336][#335]change images to thread safe

### PR DESCRIPTION
**Pull request by zhangxiaoli73**

### CODE REVIEW

**Comment by yiheng** [comment]
Can you add some new unit test for this change?

**Comment by zhangxiaoli73** [comment]
yes, add two unit test for this change.

File: dl/src/main/scala/com/intel/analytics/bigdl/dataset/image/Types.scala [hunk 1]
```diff
@@ -319,7 +319,9 @@ object BGRImage {
       val channel = fis.getChannel
       val byteArrayOutputStream = new ByteArrayOutputStream
       channel.transferTo(0, channel.size, Channels.newChannel(byteArrayOutputStream))
-      ImageIO.read(new ByteArrayInputStream(byteArrayOutputStream.toByteArray))
+      val image = ImageIO.read(new ByteArrayInputStream(byteArrayOutputStream.toByteArray))
+      if (null == image) throw new Error("Can't read file " + path)
```
Comment (yiheng) [inline dl/src/main/scala/com/intel/analytics/bigdl/dataset/image/Types.scala]: require(image != null, "error message")

File: dl/src/test/scala/com/intel/analytics/bigdl/dataset/DataSetSpec.scala [hunk 2]
```diff
@@ -229,6 +228,51 @@ class DataSetSpec extends FlatSpec with Matchers with BeforeAndAfter {
...
   }
 
+  "ImageNet data source" should "load image correct with parallel process" in {
+    val resource = getClass().getClassLoader().getResource("imagenet")
+    val labelMap = LocalImageFiles.readLabels(Paths.get(processPath(resource.getPath())))
+    val dataSet = DataSet.ImageFolder.images(Paths.get(processPath(resource.getPath())),
+      BGRImage.NO_SCALE)
+
+    val iter = dataSet.toLocal().data(train = false)
+    val parallel = 3
```
Comment (yiheng) [inline dl/src/test/scala/com/intel/analytics/bigdl/dataset/DataSetSpec.scala]: a larger parallel?

File: dl/src/test/scala/com/intel/analytics/bigdl/dataset/DataSetSpec.scala [hunk 3]
```diff
@@ -229,6 +228,51 @@ class DataSetSpec extends FlatSpec with Matchers with BeforeAndAfter {
...
+      workingThread.setDaemon(false)
+      workingThread.start()
+      workingThread
+    }).foreach(_.join())
+
+    count should be(28)
+  }
+
+  "ImageNet rdd data source" should "load image correct with parallel process" in {
+    val resource = getClass().getClassLoader().getResource("imagenet")
```
Comment (yiheng) [inline dl/src/test/scala/com/intel/analytics/bigdl/dataset/DataSetSpec.scala]: where's the parallel process?

File: dl/src/main/scala/com/intel/analytics/bigdl/dataset/image/Types.scala (hunk 1 above)
Comment (zhangxiaoli73) [inline dl/src/main/scala/com/intel/analytics/bigdl/dataset/image/Types.scala]: ok, changed

File: dl/src/test/scala/com/intel/analytics/bigdl/dataset/DataSetSpec.scala (hunk 3 above)
Comment (zhangxiaoli73) [inline dl/src/test/scala/com/intel/analytics/bigdl/dataset/DataSetSpec.scala]: useless unit test, deleted

**Comment by yiheng** [comment]
can these 3 commits merge into one?

**Comment by zhangxiaoli73** [comment]
OK, commits megred.

**Pull request merged**